src/abstractdataframe/abstractdataframe.jl

```diff
@@ -405,29 +412,92 @@ describe(df)
+                 allowed = "Allowed fields are: $allowed_fields."
+                 throw(ArgumentError(not_allowed * allowed)) 
+   end
+    # Define 4 functions for getting summary statistics 
+    # use a dict because we dont know which measures the user wants
+    function get_stats(col::AbstractArray{<:Real})
+        sumstats = summarystats(col)
+        Dict(
+            :mean => sumstats.mean,
+            :std => Compat.std(col, mean = sumstats.mean),
+            :min => sumstats.min,
+            :q25 => sumstats.q25,
+            :median => sumstats.median,
+            :q75 => sumstats.q75,
+            :max => sumstats.max,
+            :eltype => eltype(col),
+            :nmissing => nothing,
+            :nunique => nothing
+        )
+    end
+    
+    function get_stats(col::AbstractArray{<:Union{Real, Missing}})
+        nomissing = collect(skipmissing(col))
+        sumstats = summarystats(nomissing)
+        Dict(
+            :mean => sumstats.mean,
+            :std => Compat.std(nomissing, mean = sumstats.mean),
+            :min => sumstats.min,
+            :q25 => sumstats.q25,
+            :median => sumstats.median,
+            :q75 => sumstats.q75,
+            :max => sumstats.max,
+            :eltype => Missings.T(eltype(col)),
+            :nmissing => count(ismissing, col),
+            :nunique => nothing
+        )
     end
-end
 
-function StatsBase.describe(io::IO, X::AbstractVector{Union{T, Missing}}) where T
-    missingcount = count(ismissing, X)
-    pmissing = 100 * missingcount/length(X)
-    if pmissing != 100 && T <: Real
-        show(io, StatsBase.summarystats(collect(skipmissing(X))))
-    else
-        println(io, "Summary Stats:")
+    function get_stats(col::AbstractArray{>:Missing})
+        Dict(
+            :mean => nothing,
+            :std => nothing,
+            :min => nothing,
+            :q25 => nothing,
+            :median => nothing,
+            :q75 => nothing,
+            :max => nothing,
+            :eltype => Missings.T(eltype(col)),
+            :nmissing => count(ismissing, col),
+            :nunique => length(unique(col))
+        )    
+    end 
+    
+    function get_stats(col)
+        Dict(
+            :mean => nothing,
+            :std => nothing,
+            :min => nothing,
+            :q25 => nothing,
+            :median => nothing,
+            :q75 => nothing,
+            :max => nothing,
+            :eltype => eltype(col),
+            :nmissing => nothing,
+            :nunique => length(unique(col))
+        )   
+    end
+
+    # Put the summary stats into the return dataframe
+    data = DataFrame()
+    data[:variable] = names(df)
+
+    # An array of Dicts for summary statistics
+    columns = [get_stats(col) for (name,col) in eachcol(df)] 
+    for stat in stats
+        # for each statistic, loop through the columns array to find values
+        data[stat] = [columns[i][stat] for i in 1:size(df, 2)]
```

Comment:
Minor stylistic point: `[col[stat] for col in columns]` would be simpler. BTW, the comment could be moved just above `for`, and it could mention "letting the comprehension choose the appropriate type" (which is not immediately obvious).